### ISSUE TITLE
[BUG] Game visually looks laggier when streaming with OBS

### ISSUE THREAD

**Issue by thearst3rd** [MBU-Team/OpenMBU#116]
### What's going wrong

When I'm streaming with OBS, the game claims to still be running at 200+ fps, but the framerate that gets displayed on my monitor is drastically reduced, and also has occasional stutters. I'm not sure if this is even an OpenMBU bug or something with OBS, I got no idea ¯\\\_(ツ)\_/¯

Note: somehow, this _only_ happens when streaming. Recording has no effect, even though I record using much higher settings than I stream with. I really have no idea how that makes sense.

This could be an error in my OBS configuration, though I've never seen any other game act like this. I'll dig into it some more.

**Comment by CenTdemeern1** [comment]
Have you tried switching the capture mode or hook rate?

**Comment by CenTdemeern1** [comment]
I just checked it out for myself and I don't have any issues with the default settings. This might be an issue with your PC.

**Comment by CenTdemeern1** [comment]
https://www.youtube.com/watch?v=3c5COcsyGDE
Here is the test recording I did.

**Comment by CenTdemeern1** [comment]
I recorded the video in windowed mode, so that's why the resolution looks bad.
I did a test recording in fullscreen too and there are no performance changes.
https://www.youtube.com/watch?v=dv3I6MwPtvQ

**Comment by thearst3rd** [comment]
My issue happens _specifically_ when streaming. I can record at 1440p 60fps just fine, but when streaming (even though I stream at way lower resolution), the framerate noticeably dips.

The performance is still ok though, it probably at around 60fps and the game is definitely still playable. But I have a 144 hz monitor, and if I ask the game what framerate it's running at (`echo($fps::real);`), it claims it's running at 200+ fps and it is definitely not. The game stutters as well which makes it a bit more tricky to play.

Good idea about capture types/hook rates/whatever, I haven't touched any of that stuff. I'll mess around and see if something like that fixes it. It's just strange that OpenMBU is the only game on my system that acts like this.

I believe also @neolightning noticed the lagginess when he was streaming? So I don't think it's just my PC.

**Comment by AJ-Ferguson** [comment]
Is this still a problem with 94d9ff9aef00a421b28f83d08009800dc7d6d9da ?

**Comment by thearst3rd** [comment]
After doing a recording test with OBS, it seems to have much better performance than before even when using the same settings! I'll close this.

**Comment by thearst3rd** [comment]
Also, I forgot to mention this when I learned about it but the thing that was causing the issues was when I used cpu-based x264 as the encoder, rather than hardware encoding. I mentioned that it seemed laggy "specifically while streaming" but that's because I had my streaming setup to use x264 but recording with nvenc. When I set it to record using x264, I got the same lagginess even while just recording (before the new commits).
